src/main/java/org/mockito/internal/creation/instance/DefaultInstantiatorProvider.java

```diff
@@ -12,8 +12,8 @@
     private final static Instantiator INSTANCE = new ObjenesisInstantiator();
 
     public Instantiator getInstantiator(MockCreationSettings<?> settings) {
-        if (settings != null && settings.isUsingConstructor()) {
-            return new ConstructorInstantiator(settings.getOuterClassInstance());
+        if (settings != null && settings.getConstructorArgs() != null) {
```

Comment:
IIUC, this is incorrect. This check is analogous to the `isUsingConstructor()` check - the `settings` object has all sorts of properties that aren't relevant here, so even if a non-null `settings` is passed, it doesn't mean it would have constructor arguments.